src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_arm_utils.py

```diff
@@ -92,8 +108,66 @@ def _parameters(registry_name, location, storage_account_name):
     parameters = {
         'registryName': {'value': registry_name},
         'registryLocation': {'value': location},
-        'registryApiVersion': {'value': VERSION},
+        'registryApiVersion': {'value': get_acr_api_version()},
         'storageAccountName': {'value': storage_account_name},
-        'storageAccountApiVersion': {'value': '2015-05-01-preview'}
+        'storageAccountApiVersion': {'value': '2015-05-01-preview'},
+        'tenantId': {'value': get_tenant_id()}
     }
     return parameters
+
+def _arm_get_storage_account(storage_account_name):
+    '''Returns the dict of tags in the storage account.
+    :param str storage_account_name: The name of storage account
+    '''
+    result = get_resources_in_subscription('Microsoft.Storage/storageAccounts')
+    elements = [item for item in result if item.name.lower() == storage_account_name.lower()]
+
+    if len(elements) == 0:
+        return None, None
+    elif len(elements) == 1:
+        storage_account_resource_group = get_resource_group_name_by_resource_id(elements[0].id)
+        return storage_account_resource_group, elements[0].tags
+    else:
+        raise CLIError(
+            'More than one storage accounts are found with name: {}'.format(storage_account_name))
+
+def add_tag_storage_account(storage_account_name, key, value):
+    '''Add a new tag (key, value) to the storage account.
+    :param str storage_account_name: The name of storage account
+    :param str key: The key of the new tag
+    :param str value: The value of the new tag
+    '''
+    from azure.mgmt.storage.models import StorageAccountUpdateParameters
+    storage_account_resource_group, tags = _arm_get_storage_account(storage_account_name)
+
+    newKey = key
+    index = 1
+    while newKey in tags:
+        newKey = key + '_' + str(index)
+        index += 1
+        if index > 99: # Just a number to avoid infinite loops
```

Comment:
I was adding a tag (key, value) to the storage account when creating a registry, so that when we/user look at the storage account, we/user would know which registry depends on the storage account (and the other way around, we have tags in registries to show the storage account). It is not a dependency but for informational purpose, since the user can add/delete/change any of the tags.
The loop here is to avoid conflict when the same storage account is used to create multiple registries. For example, we may have 
```
{"acr":"myregistry","acr_1":"yourregistry","otherkey":"othervalue"}.
```
Making the index as the length of tags may not be enough since tags can be added/removed. For example, I may have 
```
{"acr_1":"myregistry"} 
```
and the new tag with key "acr_1" shouldn't be added.